cloud/linode/instances.go

```diff
@@ -29,176 +29,89 @@ func (e instanceNoIPAddressesError) Error() string {
 	return fmt.Sprintf("instance %d has no IP addresses", e.id)
 }
 
-func (i *instances) NodeAddresses(ctx context.Context, name types.NodeName) ([]v1.NodeAddress, error) {
-	ctx = sentry.SetHubOnContext(ctx)
-	sentry.SetTag(ctx, "node_name", string(name))
-
-	linode, err := linodeByName(ctx, i.client, name)
-	if err != nil {
-		sentry.CaptureError(ctx, err)
-		return nil, err
-	}
-
-	addresses, err := i.nodeAddresses(ctx, linode)
-	if err != nil {
-		sentry.CaptureError(ctx, err)
-		return nil, err
-	}
-
-	return addresses, nil
-}
+func (i *instances) lookupLinode(ctx context.Context, node *v1.Node) (*linodego.Instance, error) {
+	providerID := node.Spec.ProviderID
```

Comment:
I believe the sentry context needs set here.
```suggestion
	ctx = sentry.SetHubOnContext(ctx)
	providerID := node.Spec.ProviderID
```

Reply:
Note that all invocations of `lookupLinode` are preceded by the `SetHubOnContext`, so I don't think this call is needed. I'm not even sure it'd work: It's called in the parent, because `SetHubOnContext` returns a copy of said context and `lookupLinode` then has no way of returning that context to the parent.